fhirstore/fhirstore.py

```diff
@@ -159,19 +176,24 @@ def update(self, resource_type, instance_id, resource, bypass_document_validatio
 
         Returns: The updated resource.
         """
-        self.validate_resource_type(resource_type)
-
         try:
-            update_result = self.db[resource_type].replace_one(
-                {"id": instance_id},
-                resource,
-                bypass_document_validation=bypass_document_validation,
+            resource = self.normalize_resource(resource)
+        except pydantic.ValidationError as e:
+            return self.error(
+                [
+                    f"{err['msg'] or 'Validation error'}: "
+                    f"{e.model.get_resource_type()}.{'.'.join([str(l) for l in err['loc']])}"
+                    for err in e.errors()
+                ]
             )
-            if update_result.matched_count == 0:
-                raise NotFoundError
-            return update_result
-        except OperationFailure:
-            self.validate(resource)
+        update_result = self.db[resource.resource_type].replace_one(
+            {"id": instance_id},
+            json.loads(resource.json()),
+            bypass_document_validation=bypass_document_validation,
```

Comment:
This arg is not needed anymore

Plus, I think we want to remove bypass_document_validation from the `update` args

Reply:
do we want to bypass document validation at all at this point (I mean in all methods)? I'm removing it for now

Reply:
Yeah, with fhir.resources verifying all the stuff we put to/fetch from the api, I think we'll want to avoid bypassing validation most of the time.
What do you think?